packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,1202 @@
+      end: 1648826040000,
+      actions: [
+        {
+          actionType: 'VIEW',
+          timestamp: 1648826040000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {},
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: 1648825140000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {
+            title: 'title',
+            uriHash: 'TtnKwc0Lo2GY9WAi',
+          },
+        },
+      ],
+    },
+    {
+      start: 1648822380000,
+      end: 1648822620000,
+      actions: [
+        {
+          actionType: 'CLICK',
+          timestamp: 1648822500000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {
+            title: 'title',
+            uriHash: 'KXñi9EWk38wnb1tt',
+          },
+        },
+      ],
+    },
+  ],
+};
+
+describe('insight user actions preprocessing', () => {
+  describe('#mapAndSortActionsByMostRecent', () => {
+    it('should properly map and sort the raw user actions into UserAction objects', async () => {
+      const expectedMappedAndSortedActions = [
+        {
+          actionType: 'VIEW',
+          timestamp: 1648826040000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {},
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: 1648825140000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'TtnKwc0Lo2GY9WAi'},
+        },
+        {
+          actionType: 'CUSTOM',
+          timestamp: 1648822620000,
+          eventData: {type: 'useless_event', value: ''},
+          searchHub: 'community-support',
+          document: {},
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: 1648822500000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'KXñi9EWk38wnb1tt'},
+        },
+        {
+          actionType: 'SEARCH',
+          timestamp: 1648822380000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {},
+          query: '',
+        },
+        {
+          actionType: 'VIEW',
+          timestamp: 1648744500000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {},
+        },
+        {
+          actionType: 'CLICK',
+          timestamp: 1648744320000,
+          eventData: {},
+          searchHub: 'in-product-help',
+          document: {title: 'title', uriHash: 'caCgiG2JPzjZfS7G'},
+        },
+        {
+          actionType: 'CUSTOM',
+          timestamp: 1648744290000,
+          eventData: {
+            type: 'smartSnippetSuggestions',
+            value: 'expandSmartSnippetSuggestion',
+          },
+          searchHub: 'in-product-help',
+          document: {},
+        },
+      ];
+      const mockRawActions = [...fakeActions].slice(0, 8);
```

Comment:
the goal of the test didn't require the whole sample size and I was trying to limit the amount of event we iterate through so that the tests are slightly faster to run.